### PR TITLE
Download reports

### LINKED ISSUE #249: Export consultation report button
A button on the profile page that allows users to download their consultation report.
Another button for admins to download a report about all the consultations.
Depends on https://github.com/kir-dev/konzisite-api/issues/60

### PR DESCRIPTION
**Pull request by Tschonti**
Closes #249 

Added a modal where a user can download a report of their consultations. The time range of the report can be selected by the user. Admins can download admin reports, as well.

### CODE REVIEW

File: src/pages/user/components/ReportModal.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+                <Input type="date" onChange={onEndDateChange} value={formatDate(endDate)} max={formatDate(new Date())} />
+
+                <Flex justifyContent="flex-end">
+                  {endDateError && <FormHelperText color="red.500">Csak múltbeli konzikról lehet riportot generálni!</FormHelperText>}
+                </Flex>
+              </FormControl>
+              {loggedInUser?.isAdmin && (
+                <Checkbox colorScheme="green" isChecked={adminReport} onChange={(e) => setAdminReport(e.target.checked)}>
+                  Admin riport
+                </Checkbox>
```
Comment (DannySS123) [inline src/pages/user/components/ReportModal.tsx:92]: Reset the dates and the checkbox on close

File: src/pages/user/components/ReportModal.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      </Button>
+      <Modal isCentered={largeScreen} motionPreset="slideInBottom" isOpen={isOpen} onClose={onClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Riport generálása</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody pb={6}>
+            <VStack alignItems="flex-start">
+              <Text>HK-s ösztöndíjakhoz itt tudsz riport generálni az általad tartott konzultációkról.</Text>
+              {loggedInUser?.isAdmin && (
```
Comment (DannySS123) [inline src/pages/user/components/ReportModal.tsx:62]: const isAdmin = loggedInUser?.isAdmin
would be nicer, it is used in multiple places

File: src/pages/user/components/ProfileDetails.tsx [hunk 3]
```diff
@@ -88,9 +89,12 @@ export const ProfileDetails = ({ user, onLogoutPressed }: Props) => {
...
           {onLogoutPressed && (
-            <Button colorScheme="brand" rightIcon={<FaSignOutAlt />} onClick={() => onLogoutPressed()}>
-              Kijelentkezés
-            </Button>
+            <Stack direction={['column', 'column', 'row']} w={['100%', 'inherit']}>
+              <ReportModal />
+              <Button w={['100%', 'inherit']} colorScheme="brand" rightIcon={<FaSignOutAlt />} onClick={() => onLogoutPressed()}>
+                Kijelentkezés
+              </Button>
+            </Stack>
```
Comment (DannySS123) [inline src/pages/user/components/ProfileDetails.tsx:97]: With longer names it can look strange
image

**Review by DannySS123 — CHANGES_REQUESTED** [review]
Great work! 💯 
Just some minor thingies 🛠️

File: src/pages/user/components/ReportModal.tsx (hunk 1 above)
Comment (Tschonti) [reply src/pages/user/components/ReportModal.tsx:92]: This was intentional. If the user reopens the modal, they were probably dissatisfied with the report, so they want a new one, likely one that is similiar to the previous, so I think it helps if we don't reset the inputs.

However I did forget to close the modal when the report is done

**Review by DannySS123 — APPROVED** [review]

**Review by DannySS123 — APPROVED** [review]
Good enough 🏁

**Pull request merged**